Sources/Public/BacktraceQueueFileHelper.swift

```diff
@@ -0,0 +1,98 @@
+    private var minimumQueueFileSizeBytes = 4096;
+    private var maxQueueFileSizeBytes = 0
+    private var breadcrumbLogDirectory: String = ""
+        
+    public init(_ breadcrumbLogDirectory: String, maxQueueFileSizeBytes: Int) {
+        super.init()
+        self.breadcrumbLogDirectory =  breadcrumbLogDirectory
+        if (maxQueueFileSizeBytes < minimumQueueFileSizeBytes) {
+            self.maxQueueFileSizeBytes = minimumQueueFileSizeBytes;
+        } else {
+            self.maxQueueFileSizeBytes = maxQueueFileSizeBytes;
+        }
+    }
+     
+    func addInfo(_ info:[String:Any]) -> Bool {
+    
+        if let text = convertInfoIntoString(info) {
+            let textBytes = Data(text.utf8)
+            if textBytes.count > 4096 {
+                BacktraceLogger.error("We should not have a breadcrumb this big, this is a bug!")
+                return false
+            }
+            
+            let fileURL = URL(fileURLWithPath: breadcrumbLogDirectory)
+            let content = try? String(contentsOf: fileURL, encoding: .utf8)
+            do {
+                var fullContent = ""
+                if let content = content {
+                    fullContent = content
+                }
+                let contentBytes = Data(fullContent.utf8)
+                if contentBytes.count + textBytes.count > maxQueueFileSizeBytes {
+                    fullContent = ""
+                }
+                if fullContent.count == 0 {
+                    fullContent = text
+                } else {
+                    fullContent.append("\n\n\(text)")
+                }
+                try fullContent.write(to: fileURL, atomically: true, encoding: .utf8)
+                return true
+            }
+            catch {
+                BacktraceLogger.warning("\(error.localizedDescription) \nWhen ading breadcrumbs")
+                return false
+            }
+        }
+        return false
+    }
+    
+    func clear() -> Bool {
+        do {
+            let fileURL = URL(fileURLWithPath: breadcrumbLogDirectory)
+            try "".write(to: fileURL, atomically: false, encoding: .utf8)
+            return true
+        } catch {
+            BacktraceLogger.warning("\(error.localizedDescription) \nWhen clearing breadcrumbs")
+            return false
+        }
+    }
+}
+
+
+extension BacktraceQueueFileHelper {
+    
+    func getSavedContent() -> [[String:Any]]? {
+        do {
+            let fileURL = URL(fileURLWithPath: breadcrumbLogDirectory)
+            let content = try String(contentsOf: fileURL, encoding: .utf8)
+            return convertStringIntoInfo(content)
+        } catch {
+            return nil
+        }
+    }
+    
+    func convertStringIntoInfo(_ text: String) -> [[String:Any]]? {
+        if let data = text.data(using: .utf8) {
+            do {
+                return try JSONSerialization.jsonObject(with: data, options: []) as? [[String: Any]]
+            } catch {
+                return nil
```

Comment:
Shouldn't we log the exception so we know why the content couldn't be converted?